mpc-recovery/src/oauth.rs

```diff
@@ -99,13 +101,69 @@ impl IdTokenClaims {
     }
 }
 
-pub async fn get_pagoda_firebase_public_keys(
+pub async fn get_public_keys(
     client: &reqwest::Client,
-    jwt_signature_pk_url: &str,
-) -> anyhow::Result<Vec<String>> {
-    let response = client.get(jwt_signature_pk_url).send().await?;
-    let json: HashMap<String, String> = response.json().await?;
-    Ok(json.into_values().collect())
+    jwt_signature_pk_urls: &[String],
+) -> Result<Vec<String>> {
```

Comment:
It is better to create a map of keys: "provider" -> "[keys]".
Otherwise, somebody will create a token using `provider1`, but verify it with key 2.